codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -833,7 +834,8 @@ public Http2Stream closeRemoteSide() {
      */
     private final class DefaultEndpoint<F extends Http2FlowController> implements Endpoint<F> {
         private final boolean server;
-        private int nextStreamId;
+        private int nextExpectedStreamId;
```

Comment:
So let me make sure I understand the distinction of these two fields:

- `nextExpectedStreamId` is the minimum stream ID allowed for the next stream creation. This is generally equivalent to `lastStreamCreated + 1` (assuming that `lastStreamCreated` has been set).
- `nextGeneratedStreamId` is the the next stream ID to select when creating IDs.

Comments might help here.  Also I wonder if renaming could help make it more obvious:

-`nextExpectedStreamId` -> `nextStreamToCreate` (a term more closely associated with `lastStreamCreated`)
-`nextGeneratedStreamId` -> `nextStreamId` (I think the original name is probably more clear)

WDYT?